### ISSUE TITLE
custom paint class/function for dotted underline under span?

### ISSUE THREAD

**Issue by Fahad-pnw** [2dxgujun/Spannable-In-Kotlin#2]
Thank you for the helpful library? how do i add custom functions such as a paint function to draw an underline...

that is i want a dotted underline under a span? could you please guide some.

**Comment by 2dxgujun** [comment]
There is no dotted underline span in Android framework. You may need to implement your own `DottedUnderlineSpan` (check `ReplacementSpan`) then call `addSpan` to use it, for example:

```kotlin
span("Content") {
  addSpan(DottedUnderlineSpan())
  ...
}
```
or you can declare an extension function:
```kotlin
fun Span.dottedUnderline(text: CharSequence): Span = span(text) {
  addSpan(DottedUnderlineSpan())
  ...
}

val content = span {
  dottedUnderline("Content")
}
```

**Comment by Fahad-pnw** [comment]
thank you, this would help and if i do it successfully will post it here so may be if someone else searches, it could help them.
anyway, another issue i'm lately facing is applying tooltip on a subsequence/char sequence? can i give a resource id or resource tag to each of spans so i'd access them through resources so i'd apply styling on it or make it parent (anchor) of the tooltip?

**Comment by 2dxgujun** [comment]
That's great, and about your another issue, I can't understand what's your meaning, could you say it any other way or you can post some code to describe it clearly, thank you

**Comment by Fahad-pnw** [comment]
its explained better here:
https://stackoverflow.com/questions/49287807/setting-resource-id-to-spannable-in-kotlin

let me try again. 
i want to give each span in spannable / spanned a resource id/tag so i can access it as R.id.SomeSpan or cast it to a view or anything so i can pass it to a library i.e., tooltip library.

right now every tooltip is made to appear on textview as its parent, what happens is that it shows in the center of the textview and not in the center of the span. i've multiple spans that require tooltip or popup view, i want to pass either the subsequence to tooltip or popup OR pass a resource id so it shows in the center of that specific resource, i hope it makes sense now and i hope you could help me with that.

**Comment by 2dxgujun** [comment]
You can't give span a resource id, since it's some kind of text decoration and not a `View`. I think you should try to find the position of the subsequence or span in that `View` then you can use the position to show the tooltip.

**Comment by Fahad-pnw** [comment]
Thank you for guidance, could you also pls help out if you have some spare
time of how would i get the position of the subsequence? Probably
absolution position right and then pass it to a relative view?
